src/ui/alert/PairingAlert.ts

```diff
@@ -64,74 +76,103 @@ export const preparePairingAlert = async (
-               <img class="beacon-selection__img" src="${wallet.logo}"/>
-               <div class="beacon--selection__name__lg">${wallet.name}</div>
-              </div>`
-              : ''
-          }
-          `
-        }
+    listEl.appendChild(el)
+
+    const walletEl = shadowRoot.getElementById(`wallet_${walletKey}`)
+
+    const completeHandler = async (event?: KeyboardEvent) => {
+      if (event && event.key !== 'Enter') {
+        return
       }
 
-      if (walletEl) {
-        walletEl.addEventListener('click', () => completeHandler())
-        walletEl.addEventListener('keydown', completeHandler)
+      wallet.clickHandler()
+      const modalEl: HTMLElement | null = shadowRoot.getElementById('beacon-modal__content')
+      if (modalEl && type !== WalletType.EXTENSION && type !== WalletType.IOS) {
+        modalEl.innerHTML = `${
+          wallet.logo
+            ? `<p class="beacon-alert__title">Establishing Connection..</p>
+            <div id="beacon-toast-loader" class="progress-line"></div>
+            <div class="beacon--selected__container">
+             <img class="beacon-selection__img" src="${wallet.logo}"/>
+             <div class="beacon--selection__name__lg">${wallet.name}</div>
+            </div>`
+            : ''
+        }
+        `
       }
+    }
+
+    if (walletEl) {
+      walletEl.addEventListener('click', () => completeHandler())
+      walletEl.addEventListener('keydown', completeHandler)
+    }
+  }
+
+  const listContainer = document.createElement('span')
+  container.appendChild(listContainer)
+  const showWalletLists = (walletLists: PairingAlertList[]): void => {
+    listContainer.innerHTML = ''
+    walletLists.forEach((list) => {
+      const listWrapperEl = document.createElement('div')
+      listWrapperEl.classList.add('beacon-list__wrapper')
+
+      listContainer.appendChild(listWrapperEl)
+
+      const listTitleEl = document.createElement('div')
+      listTitleEl.classList.add('beacon-list__title')
+      listTitleEl.innerHTML = list.title
+      listWrapperEl.appendChild(listTitleEl)
+
+      const listEl = document.createElement('span')
+      listWrapperEl.appendChild(listEl)
+
+      list.wallets.forEach(async (wallet) => {
+        showWallet(listEl, list.type, wallet)
+      })
     })
-  })
+  }
+
+  showWalletLists(info.walletLists)
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const messageFn = async (event: any): Promise<void> => {
+    if (event.data === 'extensionsUpdated') {
+      const newInfo = await getInfo()
+      showWalletLists(newInfo.walletLists)
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const closeFn = (event: any): void => {
+    if (event.data === `closeAlert-${id}`) {
+      windowRef.removeEventListener('message', messageFn)
+    }
+  }
+
+  windowRef.addEventListener('message', messageFn)
+  windowRef.addEventListener('message', closeFn)
```

Comment:
Probably good to avoid memory leaks.

```suggestion
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  let closeFn: any => void
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  closeFn = (event: any): void => {
    if (event.data === `closeAlert-${id}`) {
      windowRef.removeEventListener('message', messageFn)
      windowRef.removeEventListener('message', closeFn)
    }
  }

  windowRef.addEventListener('message', messageFn)
  windowRef.addEventListener('message', closeFn)
```

Reply:
Thanks for catching that 😉